manifests/init.pp

```diff
@@ -93,6 +182,22 @@
   Boolean                                   $restart_on_config_change  = true,
   Boolean                                   $restart_on_package_change = true,
 ) {
+  $default_notifications_notifications = lookup('opensearch::default_notifications_notifications', Hash)
```

Comment:
If you only require those variables anywhere in config.pp you you could move them to config.pp. I like to keep the init.pp clean from unrequired evaluations.

and in config.pp you could just do:

```
$notifications_notifications_core = $opensearch::use_default_notifications_notifications_core ? {
  true  => lookup('opensearch::default_notifications_notifications_core', Hash) + $opensearch::notifications_notifications_core,
  false => $opensearch::notifications_notifications_core,
}
```

Also those files could shortened by using default for the local spacename like:

```
File {
  ensure  => file,
  owner   => 'opensearch',
  group   => 'opensearch',
  mode    => '0640',
}

file { "${config_directory}/opensearch-notifications-core/notifications-core.yml":
  content => $notifications_notifications_core.stdlib::to_yaml,
}
```

That would shorten the code a lot.

Reply:
I see no reason to keep them here, I first followed the same pattern that is used by the settings parameter (which I don't really like because it expose a bit too many things IMHO), and then switched to this one.  I did not switched the existing code to not be backwards-incompatible, but it could be a nice complement for consistency.

I will move the code, it makes sense I think that these "internal details" are not in the beginning of the main manifest but buried in the location they are actually used!